### PR TITLE
Bug 1653499 Base AMO usage stats on payload.info.addons

### PR DESCRIPTION
**Pull request by jklukas**
cc @willdurand I've created this table with one day of data (2020-07-14) for comparison.

### CODE REVIEW

File: sql/amo_prod/desktop_addons_by_client_v2/query.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        ARRAY(
+          SELECT AS STRUCT
+            REPLACE(REGEXP_EXTRACT(addon, "(.+):"), '%40', '@') AS id,
+            REGEXP_EXTRACT(addon, ":(.+)") AS version
+          FROM
+            UNNEST(SPLIT(payload.info.addons)) AS addon
+        )
+      ),
+      -- themes
+      ARRAY_AGG(STRUCT(environment.addons.theme.id, environment.addons.theme.version))
```
Comment (willdurand) [inline sql/amo_prod/desktop_addons_by_client_v2/query.sql:18]: I think themes should also be present in `payload.info.addons`.

File: sql/amo_prod/desktop_addons_by_client_v2/query.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
+WITH per_client AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    client_id,
+    sample_id,
+    ARRAY_CONCAT_AGG(
+      ARRAY(
+        SELECT AS STRUCT
+          REPLACE(REGEXP_EXTRACT(addon, "(.+):"), '%40', '@') AS id,
```
Comment (willdurand) [inline sql/amo_prod/desktop_addons_by_client_v2/query.sql:9]: I think we should also replace `%7D` with `}` and `%7B` with `{`.

Comment (jklukas) [reply sql/amo_prod/desktop_addons_by_client_v2/query.sql]: I've extracted this to a Javascript UDF where we can do proper URI decoding. It makes the query take a bit longer to run, but gets us significantly more correct results. That boosted us to 99% coverage of the v1 data.

**Comment by willdurand** [comment]
Looks good to me, I think we can move forward: https://bugzilla.mozilla.org/show_bug.cgi?id=1653499 (we'll make sure that this Telemetry field doesn't get removed by accident).

**Pull request merged**